src/status_im/contexts/chat/home/new_chat/view.cljs

```diff
@@ -52,10 +52,25 @@
 (defn contact-item-render
   [_]
   (fn [{:keys [public-key] :as item}]
-    (let [user-selected? (rf/sub [:is-contact-selected? public-key])
-          on-toggle      #(if user-selected?
-                            (re-frame/dispatch [:deselect-contact public-key])
-                            (re-frame/dispatch [:select-contact public-key]))]
+    (let [user-selected?          (rf/sub [:is-contact-selected? public-key])
+          selected-contacts-count (rf/sub [:selected-contacts-count])
+          has-reached-max-contact (cond-> selected-contacts-count
+                                    user-selected?       (dec)
+                                    (not user-selected?) (inc)
+                                    true                 (> constants/max-group-chat-contacts))
+          on-toggle               (fn []
+                                    (when has-reached-max-contact
+                                      (rf/dispatch
+                                       [:toasts/upsert
+                                        {:id   :remove-nickname
+                                         :type :negative
+                                         :text (i18n/label :t/new-group-limit
+                                                           {:max-contacts
+                                                            constants/max-group-chat-contacts})}]))
+                                    (if
+                                      user-selected?
+                                      (re-frame/dispatch [:deselect-contact public-key])
+                                      (re-frame/dispatch [:select-contact public-key])))]
```

Comment:
Hi @BalogunofAfrica, Thank you fixing this issue.

Can we use something like this?

```
selected-contacts-count (rf/sub [:selected-contacts-count])
on-toggle               (fn []
                          (if
                              user-selected?
                            (re-frame/dispatch [:deselect-contact public-key])
                            (do
                              (when (= constants/max-group-chat-contacts selected-contacts-count)
                                (rf/dispatch
                                 [:toasts/upsert
                                  {:id   :remove-nickname
                                   :type :negative
                                   :text (i18n/label :t/new-group-limit
                                                     {:max-contacts
                                                      constants/max-group-chat-contacts})}]))
                              (re-frame/dispatch [:select-contact public-key])))]
```

Reply:
- Unnecessary calculation when deselecting?
- Little confusing

Reply:
Unless I am still confused by what & why we are doing like this :thinking:

Reply:
We wouldn't benefit that much from `useCallback` here, since the component that uses it is not memoized itself

cc: @flexsurfer

Reply:
all components memoized by default we don't need to memoize them manually, so we should use use-callback or use-memo when needed

Reply:
In this context, the component only re-renders when `user-selected?` or `selected-contacts-count` changes.

And those two values are used in the callback (hence dependency), I don't think we would benefit, since `useCallback` would return a new reference anyways when those two values change.

Reply:
i don't have an opinion yet, but you're right about this context, but i feel like if we won't use hook, it might lead to not using it at all, so i would say better to have it even if won't be more effective, at least it shows that dev knew how it works when implemented it and dependencies for this function clearly visible

Reply:
I understand the concern, but applying `useMemo` or `useCallback` (by default) without context increases our memory footprint.

Reply:
> all components memoized by default we don't need to memoize them manually, so we should use use-callback or use-memo when needed

If all components are memoized by default, wouldn't passing a map like this defeat the purpose of memoization

```clj
[quo/button
     {:icon-only?          true
      :type                :grey
      :background          :blur
      :size                32
      :accessibility-label :close-scan-qr-code
      :on-press            #(rf/dispatch [:navigate-back])}
     :i/arrow-left]]
```

Since the map would get a new reference and be created anew in every render

Reply:
yes, all handlers should be moved to use-callback, this is our biggest problem that didn't strict before

Reply:
> I understand the concern, but applying `useMemo` or `useCallback` (by default) without context increases our memory footprint.

i don't think impact will be big, but not doing this will impact performance a lot, so when we realize that all devs understand how to use memoization tools, we can make it less strict

Reply:
> yes, all handlers should be moved to use-callback, this is our biggest problem that didn't strict before

I wholeheartedly disagree with this guideline to always use `useCallback`, as I explained in detail in these comments https://github.com/status-im/status-mobile/pull/19033#discussion_r1508871398 https://github.com/status-im/status-mobile/pull/19033#discussion_r1508131403

This guideline should be revisited IMO. I'll use `useCallback` a lot of times, but not always. It's literally unnecessary to always use it and it causes hard to debug issues in React apps, even when used by experienced devs (caching is hard).